### PR TITLE
Closes #16 | Create dataset loader for ALT Burmese Treebank

### LINKED ISSUE #16: Create dataset loader for ALT Burmese Treebank
Dataloader name: `alt_burmese_treebank/alt_burmese_treebank.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?alt_burmese_treebank

|     Dataset| alt_burmese_treebank |
|-------------|---|
| Description | A 20,000-sentence Burmese (Myanmar) treebank on news articles containing complete phrase structure annotation. As the final result of the Burmese component in the Asian Language Treebank Project, this is the first large-scale, open-access treebank for the Burmese language. |
| Subsets | - |
| Languages | mya |
| Tasks | Constituency Parsing |
| License | Creative Commons Attribution Non Commercial Share Alike 4.0 (cc-by-nc-sa-4.0) |
| Homepage | https://zenodo.org/records/3463010 |
| HF URL | - |
| Paper URL | https://dl.acm.org/doi/10.1145/3373268 |

### PR DESCRIPTION
**Pull request by MJonibek**
Closes #16 

Note:
PR [295](https://github.com/SEACrowd/seacrowd-datahub/pull/295) is related to this PR and issue it closes

Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @MJonibek, when trying with this command `python -m tests.test_seacrowd seacrowd/sea_datasets/alt_burmese_treebank/alt_burmese_treebank.py`, the script throws an error:

```
======================================================================
FAIL: runTest (__main__.TestDataLoader) [IDs globally unique]
Run all tests that check:
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/holylovenia/Projects/seacrowd-datahub/tests/test_seacrowd.py", line 71, in runTest
    self.test_are_ids_globally_unique(dataset_seacrowd)
  File "/Users/holylovenia/Projects/seacrowd-datahub/tests/test_seacrowd.py", line 234, in test_are_ids_globally_unique
    self._assert_ids_globally_unique(example, ids_seen=ids_seen)
  File "/Users/holylovenia/Projects/seacrowd-datahub/tests/test_seacrowd.py", line 211, in _assert_ids_globally_unique
    self._assert_ids_globally_unique(v, ids_seen)
  File "/Users/holylovenia/Projects/seacrowd-datahub/tests/test_seacrowd.py", line 219, in _assert_ids_globally_unique
    self.assertNotIn(v, ids_seen)
AssertionError: 'SNT.80188.1' unexpectedly found in {'SNT.80188.1'}
```

Everything else looks good to me based on manual check. 👍

**Comment by MJonibek** [comment]
Hi, I see, problem with unique id, I will solve this issue this week

**Comment by MJonibek** [comment]
Found the reason of this bug, during creating subnodes they have their ids, and for every new sentence id for subnodes was starting from 0 again, as the result subnodes from different sentences had same ids. Made some changes for subnodes ids: instead of id 0 it will be sent_id_0. Checked manually and with test.py, everything worked.

**Review by holylovenia — APPROVED** [review]
Thanks for your hard work and the bug explanation, @MJonibek. 👍 It makes a lot of sense. All is well during my test and review process. I'm upping the dataloader points with a +2 bonus considering the high difficulty. Thank you for tackling this dataloader.

For now, let's wait for @sabilmakbar's review.

**Comment by holylovenia** [comment]
> Thanks for your hard work and the bug explanation, @MJonibek. 👍 It makes a lot of sense. All is well during my test and review process. I'm upping the dataloader points with a +2 bonus considering the high difficulty. Thank you for tackling this dataloader.
> 
> For now, let's wait for @sabilmakbar's review.

Re-assigning the reviewer from @sabilmakbar to @SamuelCahyawijaya to give more breathing room to @sabilmakbar. 🙏

File: seacrowd/sea_datasets/alt_burmese_treebank/utils/alt_burmese_treebank_utils.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Extract id, sub_nodes and text of ROOT
+    sentence_id = sentence.split("\t")[0]
+    root_sent = sentence[sentence.find("ROOT") : -1]
+    root_subnodes = extract_parts(root_sent)
+    sub_nodes.update({i + 1: root_subnodes[i] for i in range(len(root_subnodes))})
+    sub_node_ids.extend([i + 1 for i in range(len(root_subnodes))])
+    root_text = extract_sentence(root_sent)
+
+    nodes.append({"id": f"{sentence_id+'.'+str(0)}", "type": "ROOT", "text": root_text, "offsets": [0, len(root_text) - 1], "subnodes": [f"{len(nodes)+i+1}" for i in range(len(sub_nodes))]})
```
Comment (SamuelCahyawijaya) [inline seacrowd/sea_datasets/alt_burmese_treebank/utils/alt_burmese_treebank_utils.py:49]: The subnodes naming is different to the one on `L69`

**Review by SamuelCahyawijaya — CHANGES_REQUESTED** [review]
Hi @MJonibek, thank you for implementing the dataloader.

I found one issue where the subnodes of the root node do not correspond to any node's id (they have a completely different format). I think there is a minor formatting error on `L49` of `alt_burmese_treebank.py`

File: seacrowd/sea_datasets/alt_burmese_treebank/utils/alt_burmese_treebank_utils.py (hunk 1 above)
Comment (MJonibek) [reply seacrowd/sea_datasets/alt_burmese_treebank/utils/alt_burmese_treebank_utils.py:49]: Oh, it seems I forgot to change the format on that line. Thank you, will ping you when it is fixed.

**Comment by MJonibek** [comment]
@SamuelCahyawijaya , fixed this bug

**Review by SamuelCahyawijaya — APPROVED** [review]
LGTM!

**Pull request merged**